### DISCUSSION TITLE
Is section 14.2 of MapStruct 1.5.5.Final Reference Guide actual for all Lombok version after 1.16.14

### DISCUSSION THREAD

**Discussion by AndrisAncans** [mapstruct/mapstruct#3480] [Q&A] [answered]
https://mapstruct.org/documentation/stable/reference/html/#lombok 
I'm using Lombok 1.18.28 `@Data` with or without `@Builder` annotation and MapStruct 1.5.5.Final. Don't see any difference is switching on/off `annotationProcessor "org.projectlombok:lombok-mapstruct-binding:0.2.0"` preprocessor. Is it really required?

**Comment by thunderhook** [comment]
Without knowing the technical details of the problems solved by the `lombok-mapstruct-binding` annotation processor, I can tell you this from experience:

It is not always necessary to use it. Sometimes the mappers are generated fine, sometimes not. In our experience, most problems occur when `@Value` and `@Builder` are used extensively.

For us, we always use the same Maven snippet including the `lombok-mapstruct-binding` in our 50+ projects. Years ago we did not, and after some time (due to new colleagues, refactorings, or generally added code) the problems started to occur and we had to add it anyway.

So: I would recommend using it to avoid problems in the first place. Maybe some text like this in the reference documentation would help.

**Reply by filiphr** [reply, marked as answer]
I think that we already have text saying that the `lombok-mapstruct-binding` has to be used. It is in https://mapstruct.org/documentation/stable/reference/html/#lombok

It says:

>Lombok 1.18.16 introduces a breaking change ([changelog](https://projectlombok.org/changelog)). The additional annotation processor lombok-mapstruct-binding ([Maven](https://mvnrepository.com/artifact/org.projectlombok/lombok-mapstruct-binding)) must be added otherwise MapStruct stops working with Lombok. This resolves the compilation issues of Lombok and MapStruct modules.
